docs/architecture.md

```diff
@@ -0,0 +1,222 @@
+  PRIMARY KEY (aggregate, piece)
+
+   -- We may also want to write inclusion proof here
+);
+
+CREATE INDEX inclusion_inserted_idx ON inclusion (inserted);
+
+-- Table for created aggregates. 
+CREATE TABLE aggregate
+(
+  -- commP of the aggregate (commP of the commPs)
+  link TEXT PRIMARY KEY NOT NULL,
+  -- Aggregate size
+  size number NOT NULL,
+  -- Time when the aggregate was created
+  inserted TIMESTAMP WITH TIME ZONE DEFAULT timezone('utc'::text, now()) NOT NULL
+);
+
+-- View for inclusion records that do not have an aggregate.
+CREATE VIEW cargo AS
+  SELECT *
+  FROM inclusion
+  WHERE aggregate IS NULL
+  ORDER BY inserted;
+
+-- State of aggregate deals. When aggregate is sent to spade-proxy status is 'PENDING'.
+-- When spade-proxy requests a wallet signature, status will be updated to 'SIGNED'. Once
+-- deal is accepted status changes to `APPROVED`, but if deal fails status will be set to
+-- `REJECTED`.
+CREATE TABLE deal (
+  aggregate TEXT PRIMARY KEY NOT NULL REFERENCES aggregate(link),
+  status DEAL_STATUS DEFAULT 'PENDING',
+  -- if status is an error this may contain details about the error e.g. json containing
+  -- piece CIDs that were invalid.
+  detail TEXT,
+  -- Time when aggregate was send to spade-proxy
+  inserted TIMESTAMP WITH TIME ZONE DEFAULT timezone('utc'::text, now()) NOT NULL,
+  -- Time when aggregate was signed
+  signed TIMESTAMP WITH TIME ZONE,
+  -- Time when aggregate was processed
+  processed TIMESTAMP WITH TIME ZONE
+);
+
+CREATE INDEX deal_inserted_idx ON deal (inserted);
+CREATE INDEX deal_signed_idx ON deal (signed);
+CREATE INDEX deal_signed_idx ON deal (signed);
+CREATE INDEX deal_status_idx ON deal (status);
+
+CREATE TYPE DEAL_STATUS AS ENUM
+(
+  'PENDING',
+  'SIGNED',
+  'APPROVED',
+  'REJECTED'
+);
+
+-- View of aggregates to be submitted to spade, that is all aggregates that we do not
+-- have deal records for.
+CREATE VIEW aggregate_queue AS
+  SELECT aggregate.*
+  FROM aggregate
+  LEFT OUTER JOIN deal ON aggregate.link = deal.aggregate
+  WHERE deal.aggregate IS NULL;
+
+-- View for deals pending.
+CREATE VIEW deal_pending AS
+  SELECT *
+  FROM deal
+  WHERE status = "PENDING"
+  ORDER BY inserted;
+
+-- View for deals approved by storage providers.
+CREATE VIEW deal_approved AS
+  SELECT *
+  FROM deal
+  WHERE status = "APPROVED"
+  ORDER BY processed;
+
+-- View for deals rejected by storage providers.
+CREATE VIEW deal_rejected AS
+  SELECT *
+  FROM deal
+  WHERE status = "REJECTED"
+  ORDER BY processed;
+```
+
+![DB Schema](./db-schema.png)
+https://dbdiagram.io/d/649af07402bd1c4a5e249feb
+
+## Processor Stack
+
+The w3infra processor stack manages the deployed schedulers that invoke lambda functions to consume the DB stack queues and attempt to make progress in their items to the following stages of the pipeline.
+
+Each of the schedulers might have their own schedules and can act independently of each other.
+
+The processes running in these pipeline are:
```

Comment:
I would suggest using term `workflow` in place of `process` as later tends to  different things depending on context (which is also why UCANs end up embracing workflow as a term).